subprojects/platform-jvm/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainIntegrationTest.groovy

```diff
@@ -101,6 +115,63 @@ class JavaToolchainIntegrationTest extends AbstractIntegrationSpec {
         current = (isCurrentJvm ? "current" : "non-current")
     }
 
+    def "fails when trying to change java extension toolchain spec property after it has been used to resolve a toolchain"() {
+        def jdkMetadata1 = AvailableJavaHomes.getJvmInstallationMetadata(Jvm.current())
+        def jdkMetadata2 = AvailableJavaHomes.getJvmInstallationMetadata(AvailableJavaHomes.differentVersion)
+
+        file("src/main/java/Main.java") << """
+            public class Main {}
+        """
+
+        buildScript """
+            apply plugin: "java"
+
+            java {
+                toolchain {
+                    languageVersion = JavaLanguageVersion.of(${jdkMetadata1.languageVersion.majorVersion})
+                }
+            }
+
+            compileJava {
```

Comment:
Would it be better to use a custom task here to show when the first querying happens?

Reply:
You are right, it's enough to explicitly resolve the java extension toolchain during the configuration time.